### PR TITLE
fix: [PL-27264]: Fix minor bugs in the buildcleaner script

### PR DESCRIPTION
**Pull request by gn-harness**
## Describe your changes

Changes to fix build cleaner for issues related to the maven override file.

## Checklist
- [X] I've documented the changes in the PR description.
- [X] I've tested this change either in PR or local environment.
- [X] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36186)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36186)

**Comment by gn-harness** [comment]
trigger messagecheck

**Comment by gn-harness** [comment]
trigger ti1

File: tools/build-cleaner/src/main/java/io/harness/buildcleaner/BuildCleaner.java [hunk 1]
```diff
@@ -141,11 +148,17 @@ protected Optional<BuildFile> generateBuildForModule(Path path, SymbolDependency
...
     for (String importStatement : classpathParser.getUsedTypes()) {
-      if (importStatement.startsWith("java")) {
+      if (importStatement.startsWith("java.")) {
         continue;
       }
 
       Optional<String> resolvedSymbol = resolve(importStatement, harnessSymbolMap);
+
+      // Skip the symbols from the same package.
+      if (resolvedSymbol.isPresent() && resolvedSymbol.get().substring(2).equals(path.toString())) {
```
Comment (hemanthmantri) [inline tools/build-cleaner/src/main/java/io/harness/buildcleaner/BuildCleaner.java:158]: Prefer adding a comment to explain the usage of magic number 2.

**Review by hemanthmantri — APPROVED** [review]

File: tools/build-cleaner/src/main/java/io/harness/buildcleaner/BuildCleaner.java (hunk 1 above)
Comment (gn-harness) [reply tools/build-cleaner/src/main/java/io/harness/buildcleaner/BuildCleaner.java:159]: Done

**Review by hemanthmantri — APPROVED** [review]

**Review by puneetsar — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36186)* status: Some of the checks are too old. 2 file(s) to review. 1. unresolved discussions.

**Review by gn-harness — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36186)* status: Some of the checks are too old. 2 file(s) to review. 1. unresolved discussions.

**Review by gn-harness — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36186#-:-N9Yimml4q4JsBD7qGWX:bnfp4nl)* status: Some of the checks are too old. 2 file(s) to review. 1. unresolved discussions.

**Review by puneetsar — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36186)* status: Some of the checks are too old. 2 file(s) to review. 1. unresolved discussions.

**Pull request merged**